src/AspNet.Security.OAuth.HubSpot/HubSpotAuthenticationHandler.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Globalization;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace AspNet.Security.OAuth.HubSpot;
+
+public partial class HubSpotAuthenticationHandler : OAuthHandler<HubSpotAuthenticationOptions>
+{
+    public HubSpotAuthenticationHandler(
+        [NotNull] IOptionsMonitor<HubSpotAuthenticationOptions> options,
+        [NotNull] ILoggerFactory logger,
+        [NotNull] UrlEncoder encoder,
+        [NotNull] ISystemClock clock)
+        : base(options, logger, encoder, clock)
+    {
+    }
+
+    protected override async Task<AuthenticationTicket> CreateTicketAsync(
+        [NotNull] ClaimsIdentity identity,
+        [NotNull] AuthenticationProperties properties,
+        [NotNull] OAuthTokenResponse tokens)
+    {
+        using JsonDocument userProfile = await GetUserProfileAsync(tokens);
+        var principal = new ClaimsPrincipal(identity);
+        var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, userProfile.RootElement);
+        context.RunClaimActions();
+        await Events.CreatingTicket(context);
+        return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
+    }
+
+    private async Task<JsonDocument> GetUserProfileAsync(
+        [NotNull] OAuthTokenResponse tokens)
+    {
+        using var request = new HttpRequestMessage(HttpMethod.Get, string.Format(CultureInfo.InvariantCulture, Options.UserInformationEndpoint, tokens.AccessToken));
```

Comment:
URL-encode the access token before formatting it. Also please put it into a local variable first, then just pass that variable as-is to this constructor.